modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java

```diff
@@ -65,61 +72,136 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 	}
 
-	private Object _invokeConfigurationOverride(Method method, Object[] args)
-		throws IllegalAccessException, InvocationTargetException,
-			   NoSuchMethodException {
-
-		Class<?> clazz = _configurationOverrideInstance.getClass();
-
-		method = clazz.getMethod(method.getName(), method.getParameterTypes());
+	private Object _getValue(
+			Class<?> returnType, String key, boolean getReturnTypeDefault)
```

Comment:
Boolean params should not look like a method name. Probably a better name would be "returnTypeDefault".

What I would like us to think through though is to avoid having this "Control Parameter" if possible, since this type of control params are very error prone. Maybe even consider two different methods if necessary (getValue and getValueOrNull, or something like that). What do you think?